src/utils.js

```diff
@@ -262,75 +259,83 @@ export default function getPossibleRequests(
   ];
 }
 
-const matchCss = /\.css$/i;
+function promiseResolve(callbackResolve) {
+  return (context, request) =>
+    new Promise((resolve, reject) => {
+      callbackResolve(context, request, (err, result) => {
+        if (err) reject(err);
+        else resolve(result);
+      });
+    });
+}
```

Comment:
I think we need to solve it on webpack side, but it is out of scope this PR, so let's do it in future

Reply:
Yeah I had to do this to be compatible with `enhanced-resolve`. `loaderContext.getResolve()` [returns a promise-based function](https://github.com/webpack/webpack/blob/v4.44.1/lib/NormalModule.js#L205), but `enhanced-resolve` does not, unfortunately.